src/Swift.Bindings/src/Model/TypeSpec.cs

```diff
@@ -0,0 +1,380 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text;
+
+namespace BindingsGeneration;
+
+public abstract class TypeSpec
```

Comment:
If you want, I'm happy to move it. I don't actually care that much. In terms of the difference, a `*Decl` is a declaration of a type which includes all of its members. A `TypeSpec` is a specification of a type with no information about members (etc). You would use A `TypeSpec` within a `*Decl` to label the types of its members.
For example, right now an `ArgumentDecl` has a `TypeIdentifier` which is a `TypeDecl`, but that really should be a `TypeSpec`.

From a modeling point of view:
``` swift
public struct Foo { // Foo is a StructDecl because it *declares* the type Foo
    public var X: (Int, Bool) // and x is a FieldDecl because it *declares* the field X,
                                            // however (Int, Bool) is a type specification because it specifies the type of the field,
                                            // but includes no further information
}